linter/rules/primary_key_is_not_first_dimension_in_view.py

```diff
@@ -0,0 +1,17 @@
+from linter.rule import Rule
+from typing import Any, Tuple
+
+
+class PrimaryKeyIsNotFirstDimensionInView(Rule):
+    def applies_to() -> Tuple[str, ...]:
+        return ('view',)
+
+    def run(self, view: Any) -> bool:
+        dimensions = view.get('dimensions', [])
+        primary_key_is_first = True
+        if len(dimensions) > 0:
+            primary_key_is_first = False
+            first_dimension = dimensions[0]
+            if 'primary_key' in first_dimension:
+                primary_key_is_first = True
+        return primary_key_is_first
```

Comment:
With lkml, dimensions are read in the order they're defined in a LookML file. Thus, we can ensure the primary key is the first **dimension** in a file. 

The limitation of this is that each field type (dimension, dimension_groups, and measures) is read in independently, so I don't think it's possible to compare dimension ordering to measure ordering. 

For example, if the measures are defined above the dimensions in a file but the first dimension is a primary key, this rule won't raise a warning. Even though the measures are above the dimensions.

Reply:
I think that should be fine! Good catch.